synapse/storage/stats.py

```diff
@@ -103,6 +193,26 @@ def update_room_state(self, room_id, fields):
             desc="update_room_state",
         )
 
+    @cached()
+    def get_earliest_token_for_stats(self, stats_type, id):
```

Comment:
Does this cache not need to be invalidated?

Reply:
I believe I have addressed this — but please can you check this specifically before resolving this comment? I feel like it's something that could easily bite in the bum if it's not correct.